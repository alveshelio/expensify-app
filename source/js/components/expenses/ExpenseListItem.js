import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { removeExpense } from '../../actions/expensesActions';

const ExpenseListItem = ({
  id,
  description,
  amount,
  note,
  createdAt,
  sortBy,
  text,
  dispatch,
  history,
}) => {
  return (
    <div className='expense-list-item'>
      <span>Description: {description}</span>
      {note && <span>Note: {note}</span>}
      <span>Amount: {amount}</span>
      {createdAt && <span>Created On: {createdAt.format('MMM Do, YYYY')}</span>}
      <button onClick={() => dispatch(removeExpense(id))}>Remove</button>
      <Link to={`/edit/${id}`}>Edit</Link>
      <hr />
    </div>
  );
};

ExpenseListItem.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  note: PropTypes.string,
  createdAt: PropTypes.shape({

  }),
  sortBy: PropTypes.string,
  text: PropTypes.string,
};

// const mapStateToProps = state => ({
//   expenses: state.expenses,
// });

export default connect()(ExpenseListItem);
