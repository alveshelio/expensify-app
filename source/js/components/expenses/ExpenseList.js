import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../../selectors/expensesSelectors';
import { removeExpense } from '../../actions/expensesActions';

export const ExpenseList = ({ expenses, filters, dispatch }) => {
  return (
    <div>
      <h1>ExpenseList</h1>
      {
        expenses.length > 0 ?
          expenses.map(expense =>
            (<ExpenseListItem
              key={expense.id}
              {...filters}
              {...expense}
              onRemoveHandle={id => dispatch(removeExpense(id))}
            />)) :
          <p>You don&apos;t have any expenses yet</p>
      }
    </div>
  );
};

ExpenseList.propTypes = {
  expenses: PropTypes.arrayOf(PropTypes.shape({
    description: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    note: PropTypes.string,
    createdAt: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  filters: PropTypes.shape({
    text: PropTypes.string,
    sortBy: PropTypes.string,
    startDate: PropTypes.number,
    endDate: PropTypes.number,
  }),
  dispatch: PropTypes.func,
};

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);
