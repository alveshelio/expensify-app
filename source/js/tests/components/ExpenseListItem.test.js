/* eslint no-undef: "off" */

import React from 'react';
import { shallow } from 'enzyme';

import ExpenseListItem from '../../components/expenses/ExpenseListItem';
import expenses from '../../fixtures/expenses';

test('it should render ExpenseListItem with an expense', () => {
  const wrapper = shallow(<ExpenseListItem
    {...expenses[0]}
    onRemoveHandle={() => console.log('removed')}
  />);

  expect(wrapper).toMatchSnapshot();
});
