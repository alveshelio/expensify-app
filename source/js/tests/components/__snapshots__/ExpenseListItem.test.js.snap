// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render ExpenseListItem with an expense 1`] = `
<div
  className="expense-list-item"
>
  <span>
    <Link
      replace={false}
      to="/edit/1"
    >
      Watter Bill
    </Link>
  </span>
  <span>
    Note: 
    The watter is expensive
  </span>
  <span>
    Amount: 
    45.54
  </span>
  <span>
    Created On: 
    Oct 13th, 2017
  </span>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
  <hr />
</div>
`;
